File: Day-Four/javascript/Check-Out_App/CheckOutFunction.js
function getSubTotal(itemPriceList){

    let subTotal = 0.0
    for(let price of itemPriceList){
        subTotal += price
    
    }
    return subTotal
}

module.exports = { getSubTotal }

function getRate(rate){

    let ratep = rate/100
    return ratep    
}

module.exports = { getSubTotal, getRate}

function getDiscount(itemPriceList, rate){

    let subTotal = getSubTotal(itemPriceList)
    rate = getRate(rate)
    let discount = subTotal * rate 
    
    return discount
}
module.exports = { getSubTotal, getRate, getDiscount}

function getVat(itemPriceList){

    let subTotal = getSubTotal(itemPriceList)
    let vat = subTotal * 0.175
    vat = Math.round(vat, 2)

    return vat
}

module.exports = { getSubTotal, getRate, getDiscount, getVat}

function getBillTotal(itemPriceList, discountRate){

    let subTotal = getSubTotal(itemPriceList)

    let discount = getDiscount(itemPriceList, discountRate)

    let vat = getVat(itemPriceList)
    let totalBill = subTotal - discount + vat
    return totalBill
}

module.exports = { getSubTotal, getRate, getDiscount, getVat, getBillTotal}

function getChange(deposit, itemPriceList, discountRate){
    let change = 0
    let totalBill = getBillTotal(itemPriceList, discountRate)
    if (totalBill > deposit){
        change = totalBill - deposit
    }
    
    else{
        change = deposit - totalBill
    }
    return change;
}

module.exports = { getSubTotal, getRate, getDiscount, getVat, getBillTotal, getChange }

File: Day-Four/python/CheckOut/check_out_function.py
#Funtions
import math

def get_sub_total(item_price_list):
    sub_total = 0.0
    for price in item_price_list:

        sub_total += price

    return sub_total


def get_rate(rate):

    ratep = rate/100
    return ratep

def get_discount(item_price_list, rate):

    sub_total = get_sub_total(item_price_list)

    rate = get_rate(rate)

    discount = sub_total * rate
    return discount

def get_vat(item_price_list):

    sub_total = get_sub_total(item_price_list)
    vat = sub_total * 0.175
    vat = round(vat, 2)

    return vat

def get_bill_total(item_price_list, discount_rate):
    sub_total = get_sub_total(item_price_list)

    discount = get_discount(item_price_list, discount_rate)
    vat = get_vat(item_price_list)

    total_bill = sub_total - discount + vat
    return total_bill

def get_change(deposit, item_price_list, discount_rate):

    total_bill = get_bill_total(item_price_list, discount_rate)
    if total_bill > deposit:
        change = total_bill - deposit
    else:
        change = deposit - total_bill

    return change
